client/post-editor/media-modal/detail/detail-item.jsx

```diff
@@ -84,6 +86,69 @@ class EditorMediaModalDetailItem extends Component {
 		);
 	}
 
+	handleOnRestoreClick = () => {
+		const { site, item, onRestore } = this.props;
+		onRestore( site && site.ID, item );
+	}
+
+	renderRestoreButton() {
+		const {
+			item,
+			translate
+		} = this.props;
+
+		//do a simple guid vs url check
```

Comment:
Right we can only detect whether an image has *never* been edited, but I think it's still useful. We can fix this a bit more eloquently for not displaying after a restore in a follow up PR.